Setup/Patch/Data/UpdateOldPrivateKeysToEncryptedVersions.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Klaviyo\Reclaim\Setup\Patch\Data;
+
+use Klaviyo\Reclaim\Helper\ScopeSetting;
+
+use Magento\Framework\Setup\ModuleDataSetupInterface;
+use Magento\Framework\Setup\Patch\DataPatchInterface;
+use Magento\Framework\Encryption\EncryptorInterface;
+use Magento\Framework\Setup\Patch\PatchVersionInterface;
+
+/**
+ * Patch is mechanism, that allows to do atomic upgrade data changes
+ */
+class UpdateOldPrivateKeysToEncryptedVersions implements DataPatchInterface, PatchVersionInterface
+{
+    /**
+     * Klaviyo ScopeSetting
+     * @var ScopeSetting $_klaviyoScopeSetting
+     */
+    protected $_klaviyoScopeSetting;
+
+    /**
+     * @var ModuleDataSetupInterface $moduleDataSetup
+     */
+    private $moduleDataSetup;
+
+    /**
+     * Magento Encryptor
+     * @var EncryptorInterface $_encryptor
+     */
+    protected $_encryptor;
+
+    /**
+     * @param ScopeSetting $_klaviyoScopeSetting
+     * @param ModuleDataSetupInterface $moduleDataSetup
+     * @param EncryptorInterface $encryptor
+     */
+    public function __construct(
+        ScopeSetting $klaviyoScopeSettings,
+        ModuleDataSetupInterface $moduleDataSetup,
+        EncryptorInterface $encryptor
+    )
+    {
+        $this->_klaviyoScopeSetting = $klaviyoScopeSettings;
+        $this->moduleDataSetup = $moduleDataSetup;
+        $this->_encryptor = $encryptor;
+    }
+
+    /**
+     * Do Upgrade
+     *
+     * @return void
+     */
+    public function apply()
+    {
+        $this->moduleDataSetup->getConnection()->startSetup();
+        $this->updateOldPrivateKeysToEncryptedVersions();
+        $this->moduleDataSetup->getConnection()->endSetup();
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function getAliases()
+    {
+        return [];
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public static function getDependencies()
+    {
+        return [];
+    }
+
+    public static function getVersion()
+    {
+        return '1.1.7';
```

Comment:
I see the docstring for the method below mentions something about version 1.1.7 which I assume is what this corresponds to. If so is there any other info you can add int eh docstring for how this is utilized? Or does this version number need to be updated?

Reply:
The Patch script will only run if the version is less than what is specified in this method. The context below is all that is needed I suppose. If we need to make new changes to any Data, we will require to create a new patch script in the directory corresponding to the version just before the release planned for the patch script

Reply:
This is more of a backward compatibility thing required by Magento for migration to Declarative schema